doc/digital-receipt/implementation/digital-receipt-implementation.md

```diff
@@ -0,0 +1,667 @@
+
+<details>
+<summary>The following example shows how to extract the value of a flag into the ftState property.</summary>
+  
+```
+if ((ReceiptResponse.ftState & 0x4154000000000001) != 0) 
+{ 
+    //your code in case of out of service condition 
+}
+if ((ReceiptResponse.ftState & 0x4154000000000004) != 0)
+ { 
+    //your code in case of SSCD permanently out of service condition
+ }
+```
+</details>
+
+## Germany 
+
+In the event of a failure or disruption of the internet connection, we recommend to print the receipt and make it available to the customer. If you receive the ftState 0x4154000000000001 or 0x4154000000000004 as a ReceiptResponse, no digital receipt should be visualized as a QR-Code or scanned as Give-Away - the receipt should be printed out. The country-specific code is made of the country's code value following the ISO-3166-1-ALPHA-2 standard, converted from ASCII into hex. For Germany (DE) this is 0x4445, which results in 0x4445000000000002 as the value for the "security mechanism was not able to communicate with the TSE device for at least one cycle" status.
+
+| Value  | Description | Middleware value |
+| ------------- | ------------- | ------------- |
+| 0x4445000000000002  | The security mechanism was not able to communicate with the TSE device for at least one cycle. If this is the case, no more communication attempts are done to avoid long waiting times for each Receipt request/Receipt response sequence. To leave this state, a Zero-Receipt must be sent, which forces a communication retry towards the TSE device. Receipts created in a state where no communication is possible with the TSE device are protected by the security mechanism of fiskaltrust.  | 1.0  |
+| 0x4445000000000100  | The Middleware is in the process of switching SCUs. This state is returned in case any receipts are processed between the initialize-switch receipt and the finish-switch receipt. These receipts are protected by the fiskaltrust.SecurityMechanism, but not sent to any TSE, as no SCU is connected at this point.  | 1.3.19  |
+
+The following example shows how to extract the value of a flag into the ftState property.
+
+<details>
+<summary>The following example shows how to extract the value of a flag into the ftState property.</summary>
+
+```
+if ((ReceiptResponse.ftState & 0x4445000000000002) != 0)
+{ 
+    //your code in case of out of service condition 
+}
+if ((ReceiptResponse.ftState & 0x4445000000000100) != 0)
+ { 
+    //your code in case of SSCD permanently out of service condition
+ }
+```
+</details>
+
+# Mandatory fields for digital receipt visualization 
+
+This chart shows the required data fields to visualize the hole dataset of the digital receipts properly, like the services or items sold, payment methods like voucher or card payment transaction data from your PSP. 
```

Comment:
Also, is it really a good idea to put this here? This is not specifically required for the digital receipt, but mandatory if you want the Middleware to properly work.. e.g. in Germany, the DSFInV-K will be incorrect if you don't transmit those values correctly (and the MW will anyway throw if some aren't, because they are hard-mandatory)